src/toil/jobStores/fileJobStore.py

```diff
@@ -244,8 +244,11 @@ def jobs(self):
                 logger.warning('Job Dir: %s' % i)
                 if i.startswith(self.JOB_DIR_PREFIX):
                     # This is a job instance directory
+                    jobId = self._getJobIdFromDir(os.path.join(tempDir, i))
                     try:
-                        yield self.load(self._getJobIdFromDir(os.path.join(tempDir, i)))
+                        if self.exists(jobId):
+                            yield self.load(jobId)
+                        raise NoSuchJobException(jobId)
```

Comment:
Now it's going to raise and immediately catch this exception for every job, whether it exists and got yielded or not. It shouldn't do that.